bundles/org.openhab.binding.mqtt.homeassistant/src/test/java/org/openhab/binding/mqtt/homeassistant/internal/component/SensorTests.java

```diff
@@ -77,6 +77,9 @@ public void test() throws InterruptedException {
                 .build().getPattern(), is("%s %unit%"));
 
         waitForAssert(() -> assertState(component, Sensor.SENSOR_CHANNEL_ID, UnDefType.UNDEF), 10000, 200);
+
+        publishMessage("zigbee2mqtt/bridge/state", "{ \"state\": \"offline\" }");
```

Comment:
It seems that thing should be offline after receiving the message.
Probably the status tracker is async or replaced by mock in the abstract test.

Reply:
🤦  copy/paste error. good catch. and yeah, I had to wrap it in waitForAssert to get it to pass.

Reply:
it ended up a little more complicated than that. Since the ThingHandlerCallback is a mock, it wasn't actually propagating the OFFLINE status to the Thing. Now fixed.

Reply:
Great!
You could make an cosmetic improvement by replacing inline Answer class implementation by Java lambda: `doAnswer(invocation -> {...})...`

Reply:
thanks! java definitely isn't my primary language, so it's always good to get little tips like this to continue to improve in it